lib/auth/touchid/authenticate.m

```diff
@@ -0,0 +1,61 @@
+// go:build touchid
+//  +build touchid
+
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "authenticate.h"
+
+#import <CoreFoundation/CoreFoundation.h>
+#import <Foundation/Foundation.h>
+#import <Security/Security.h>
+
+#include "common.h"
+
+int Authenticate(AuthenticateRequest req, char **sigB64Out, char **errOut) {
```

Comment:
If we're not actually going to distinguish errors with different error codes, why not just return the error as a `char *` and let a null pointer indicate no errors? (same in `Register`)

Reply:
I wanted to give myself room for an eventual `OSStatus`-like integer return. It's not obvious to me which underlying APIs I may end up calling, and not all of them return CFErrorRef/NSError/error types with string messages.

Since there is still some work to land here I'd rather keep the signatures, even if they are a bit redundant.